app/controllers/redirect/return_to_sp_controller.rb

```diff
@@ -5,9 +5,12 @@ class ReturnToSpController < Redirect::RedirectController
     before_action :validate_sp_exists
 
     def cancel
-      redirect_url = sp_return_url_resolver.return_to_sp_url
-      analytics.return_to_sp_cancelled(redirect_url: redirect_url, **location_params)
-      redirect_to(redirect_url, allow_other_host: true)
+      @redirect_url = sp_return_url_resolver.return_to_sp_url
+      analytics.return_to_sp_cancelled(redirect_url: @redirect_url, **location_params)
```

Comment:
We should probably not log this event until the user is actually redirected to the partner application. Currently it's being logged when they view the confirmation screen.

As mentioned when we were pairing, I'm wondering if it'd be better to create a separate controller/action and leave the redirect controller responsible only for redirecting, not for showing any user-facing views. That way, we can keep the analytics here but assume they'd only reach this action when it's time for them to be redirected.

Reply:
> We should probably not log this event until the user is actually redirected to the partner application. Currently it's being logged when they view the confirmation screen.

Additionally, we may want a separate logging event for when the user visits the confirmation screen.

Reply:
That makes sense.

I remember at the very end of our pair session that we could try using a query parameter to redirect and that is why I explored that route. I'll go back to what we first talked about.